proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,257 @@
+function interaction.] Indeed, it is expected that no serialization is required
+when invoking a function via Interface Types' adapters.
+
+### Language Specific Module Systems
+
+Interface Types are _not_ intended to replacement language specific module
+systems. For example, C/C++ modules would continue to be linked together into
+single webAssembly modules. 
+
+## WebIDL
+
+[Web IDL][WebIDL] is clearly an important IDL for Web APIs. And, since the web
+platform is a crucial domain for webAssembly, it seems necessary to address the
+question of "why not WebIDL" for describing APIs in Interface Types.
+
+The case against WebIDL is comprised of three principal arguments:
+
+* WebIDL is too strongly aligned to JavaScript;
+* WebIDL has features that are not relevant in the context of webAssembly; and
+* WebIDL does not support certain necessary features.
+
+### Javascript
+
+WebAssembly can be characterized as a vehicle that allows languages _other than_
+Javascript to be executed on the web (specifically, in a browser tab).
+
+On the other hand, WebIDL is designed to be an IDL for Javascript. This results
+in WebIDL having features that may be problematic for other languages.
+
+For example, WebIDL has the concepts of `object` and _Indexed Properties_; where
+`object` means the value is a Javascript object and an _Indexed Property_ is one
+which exposes an interface that allow indexed access.
+
+Many languages do not support these concepts; and it may be quite onerous to do
+so. However, any feature of an IDL should be useable by any API designer to
+create an appropriate API. Features of an IDL that target Javascript sould not
+be used by an API designer that wishes to design for many different languages.
+
+There is a fine distinction to be made in making 'avoidance' recommendations and
+constructing a completely new IDL.
+
+#### WebIDL Strings
+
+Certain aspects of WebIDLs design seem to be somewhat overelaborated. For
+example, there are two variants of floating point numbers: float and
+unrestricted floats. (With corresponding equivalents for double precision
+floating point numbers.) The difference between a `float` and an `unrestricted
+float` in WebIDL is one that is not common in programming languages.
+
+Similarly,  WebIDL  has   three  variants  of  string   type:  the  `DOMString`,
+`ByteString` and `USVString` string. Again, the distinction between `DOMString`s
+and `USVString`s is not one that is common in programming languages.
+
+### Extended Attributes
+
+Heavy use is made in WebIDL of so-called extended attributes. For example, the
+`AllowShared` extended attribute is intended to mark a buffer type to be backed
+by a `SharedArrayBuffer`.
+
+Apart from the obvious dependence on a specific feature of Javascript (shared
+array buffers), allowing sharing on values contradicts one of the key design
+criteria for Interface Types -- namely supporting interoperability in a cross
+ownership domain where there is no sharing.
```

Comment:
Here I begin to see why we started with the discussion of ownership, but my feeling is that digging this deep into a particularly baroque extended attribute is missing the forest for the trees. The next two paragraphs do a nice job of explaining a more general reason why WebIDL's extended attributes are problematic for the Interface Types use case, without opening up discussions about exactly that `AllowShared` means, how it relates to the shared-nothing linking use-case, etc.

I'd recommend focusing on the following two paragraphs instead and removing this detour into details.

Reply:
I used AllowShared because it was literally the first extended attribute that shows up in the WebIDL specification. That it tied in nicely with ownership was a bonus...
It is also an example of a heavily JavaScript centric concept. I will think about refactoring the text though.

Reply:
I would disagree that it's "heavily JavaScript"; it's really just "heavily SharedArrayBuffer", which means it's probably only well-known to 5-10 people on Earth. I think something like [Unforgeable](https://heycam.github.io/webidl/#Unforgeable) would be a better choice, something that is really JS-specific.

Reply:
I can definitely go with unforgeable.

Reply:
Also, amusingly, once we have some sort of `slice`/`view` interface type, I think we'll want that slice/view type to reflect whether the underlying memory type is [shared](https://webassembly.github.io/threads/syntax/types.html#memory-types) or not, so `AllowShared` isn't 100% wrong either.

Reply:
Unless we could find a way of guaranteeing that a slice would never be corrupted, I remain opposed to introducing it.

Reply:
I still think we shouldn't fixate on `AllowShared`, but some other attribute.  The point of a `slice` is that it is completely clobbered by being written into so it can only be "corrupted" insofar as one can return a "corrupted" return value, which is to say that it is no worse.  I think the more important thing is that a `slice` can't be held onto longer than the duration of a call, as we've discussed in previous wasm calls.

Reply:
Rewritten to use unforgeable